src/status.js

```diff
@@ -0,0 +1,137 @@
+      }
+    });
+  });
+};
+
+const generateTasks = (arr) => {
+  arr.forEach((task) => {
+    const li = document.createElement('li');
+    li.classList.add('listEl');
+    const input = document.createElement('input');
+    const label = document.createElement('label');
+    const edited = document.createElement('span');
+    const deleteBtn = document.createElement('button');
+    deleteBtn.type = 'button';
+    deleteBtn.innerText = 'X';
+    deleteBtn.classList.add('editBtn');
+    deleteBtn.classList.add('deleteBtn');
+    edited.innerText = '✓';
+    edited.classList.add('red');
+    edited.classList.add('editBtn');
+    input.type = 'checkbox';
+    input.classList.add('checkbox');
+    label.setAttribute('contenteditable', true);
+    if (task.completed === true) (input.checked = true); else (input.checked = false);
+    label.textContent = task.description;
+    li.appendChild(input);
+    li.appendChild(label);
+    li.appendChild(deleteBtn);
+    li.appendChild(edited);
+    taskContainer.appendChild(li);
+    const checkbox = document.querySelectorAll('.checkbox');
+    checkbox.forEach((box, i) => {
+      box.addEventListener('click', () => {
+        if (box.checked === true) {
+          arr[i].completed = true;
+        } else {
+          arr[i].completed = false;
+        }
+        localStorage.setItem('tasks', JSON.stringify(tasks));
+      });
+    });
+    editDescription();
+  });
+  deleteTask(arr);
+  clearAll(arr);
+};
+
+addTask.addEventListener('keypress', (e) => {
+  if (addTask.value !== '' && e.key === 'Enter') {
+    if (localStorage.length !== 0) {
+      const storedTasks = JSON.parse(localStorage.getItem('tasks'));
+      const task = new Task(addTask.value, false, storedTasks.length);
+      storedTasks.push(task);
+      localStorage.setItem('tasks', JSON.stringify(storedTasks));
+      clearCont();
+      generateTasks(storedTasks);
+      checkBox(storedTasks);
+    } else {
+      const task = new Task(addTask.value, false, tasks.length);
+      tasks.push(task);
+      localStorage.setItem('tasks', JSON.stringify(tasks));
+      clearCont();
+      generateTasks(tasks);
+      checkBox(tasks);
+    }
+  }
+});
+
+const deleteTask = (arr) => {
+  const deleteBtns = document.querySelectorAll('.deleteBtn');
+  deleteBtns.forEach((button, i) => {
+    button.addEventListener('click', () => {
+      if (localStorage.length !== 0) {
+        const storedTasks = JSON.parse(localStorage.getItem('tasks'));
+        storedTasks.splice(i, 1);
+        storedTasks.forEach((task, i) => {
+          task.index = i;
+        });
+        localStorage.setItem('tasks', JSON.stringify(storedTasks));
+        clearCont();
+        generateTasks(storedTasks);
+      } else {
+        arr.splice(i, 1);
+        arr.forEach((task, i) => {
+          task.index = i;
+        });
+        localStorage.setItem('tasks', JSON.stringify(arr));
+        clearCont();
+        generateTasks(arr);
+      }
+    });
+  });
+};
```

Comment:
- [ ] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique (currently they maintainer their generated index).

Reply:
Hi @omar-labana, thank you for your feedback 🙏 ! 
I'm not entirely sure as to what I should do as the index is actually being updated in lines 106/107 and 114/115. Let me share some screenshots below. Is it ok like this or do I need to change anything about it?
Thank you for confirming! 

<img width="530" alt="Screen Shot 2021-12-03 at 7 41 09" src="https://user-images.githubusercontent.com/75140420/144612470-e521d5b9-1423-4384-865e-92b60ba8deb9.png">
<img width="499" alt="Screen Shot 2021-12-03 at 7 41 18" src="https://user-images.githubusercontent.com/75140420/144612473-551724cf-fa93-4f3e-93ec-89f7ec3be5fb.png">
<img width="532" alt="Screen Shot 2021-12-03 at 7 41 29" src="https://user-images.githubusercontent.com/75140420/144612479-169521dd-e78b-4540-b7d1-5b4162ae71a3.png">
?